### PR TITLE
Dynamic DID Resolution

### PR DESCRIPTION
**Pull request by decentralgabe**

### CODE REVIEW

**Comment by decentralgabe** [comment]
cc: @andorsk reworked some of what you contributed

File: did/did_test.go [hunk 1]
```diff
@@ -3,18 +3,67 @@ package did
...
+	assert.NotEmpty(t, doc)
+
+	// did peer
+	didPeer := "did:peer:0z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"
+	doc, err = resolver.Resolve(didPeer)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, doc)
+
+	// did web
+	gock.New("https://demo.ssi-sdk.com").
```
Comment (nitro-neal) [inline did/did_test.go:60]: awesome

**Review by nitro-neal — APPROVED** [review]
Looks good!

**Comment by andorsk** [comment]
Nice! Thanks @decentralgabe. LGTM as well.

**Pull request merged**